File: 0011-container-with-most-water/0011-container-with-most-water.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        res=0
        l=0
        r=len(height)-1
        while(l!=r):
            area=min(height[l],height[r])*(r-l)
            res=max(area,res)
            if(height[r]>=height[l]):
                l=l+1
            else:
                r=r-1
        return res



        

File: 0011-container-with-most-water/0011-container-with-most-water.cpp
class Solution {
public:
    int maxArea(std::vector<int>& height) {
        int res = 0;
        int l = 0;
        int r = height.size() - 1;

        while (l < r) {
            int area = std::min(height[l], height[r]) * (r - l);
            res = std::max(res, area);

            if (height[r] >= height[l]) {
                l++;
            } else {
                r--;
            }
        }
        
        return res;
    }
};